nxc/protocols/ldap.py

```diff
@@ -488,6 +494,270 @@ def kerberos_login(self, domain, username, password="", ntlm_hash="", aesKey="",
                 )
                 return False
 
+    def create_ldap3_connection(self):
+        """
+        Return an ldap3.Connection.
+
+        - NTLM bind requires a plaintext password.
+        - If LM/NT hashes are provided (no password), switch to Kerberos automatically.
+        - Kerberos path: try LDAP:389, fallback to LDAPS:636 on strongerAuthRequired.
+        """
+        target = self.host
+        domain = (self.domain or "").upper()
+        user   = self.username or ""
+        pwd    = getattr(self, "password", "") or ""
+        lmhash = getattr(self, "lmhash", "") or ""
+        nthash = getattr(self, "nthash", "") or ""
+        aeskey = getattr(self, "aesKey", "") or ""
+        kdc    = getattr(self, "kdcHost", None)
+        port   = int(getattr(self.args, "port", 389))
+        port_explicit = bool(getattr(self.args, "port_explicitly_set", False))
+
+        explicit_k = bool(getattr(self.args, "kerberos", False))
+        hash_only  = (not pwd) and (bool(nthash) or bool(lmhash))
+        aes_only   = (not pwd) and bool(aeskey)
+        use_k      = explicit_k or hash_only or aes_only
+        use_cc     = bool(getattr(self.args, "use_kcache", False))
+
+        if hash_only and not explicit_k:
+            self.logger.display("No -k supplied: switching to Kerberos because ldap3 NTLM doesn’t support hash-only auth.")
+
+        if not use_k:
+            def _ntlm_bind(ssl: bool, bind_port: int):
+                try:
+                    self.logger.extra["protocol"] = "LDAPS" if ssl else "LDAP"
+                    self.logger.extra["port"] = str(bind_port)
+                    server = ldap3.Server(
+                        target,
+                        port=bind_port,
+                        use_ssl=ssl,
+                        get_info=ldap3.NONE,
+                    )
+                    user_ntlm = f"{domain}\\{user}" if domain else user
+                    conn = ldap3.Connection(
+                        server,
+                        user=user_ntlm,
```

Comment:
why not use `self` ?

self.username ?
self.password ?
self.kerberos ?
seff.ccache ?

All these variables (not conventional to nxc) are adding a complexity to the function which is not necessary.